collection/sites/mastodon.yml

```diff
@@ -0,0 +1,18 @@
+name: "Mastodon"
+host: mastodon.social
+example: https://mastodon.social/@HackerNewsBot/100181134752056592
+extract: "!//(?'host'[-.\\w]+)/@(?'name'\\w+)/(?'id'\\d+)!"
+oembed:
+    endpoint: https://mastodon.social/api/oembed
+    scheme: https://mastodon.social/@{@name}/{@id}
+scrape:
+    - extract:
+        url: "!https://(?'host'[-.\\w]+)/@(?'name'\\w+)/(?'id'\\d+)!"

```

Comment:
@JoshyPHP could you please judge if this part of definition is fully correct, especially the inported part of `extract` which looks like `'extract'=>['#"url":"https://(?\'host\'[-.\\w]+)/@(?\'name\'\\w+)/(?\'id\'\\d+)"#']` in `CachedDefinitionCollection.php`.

Reply:
At first glance it looked fine but on a second read I realized it was a trick of the eye caused by the YML structure. The regexp I use is `#"url":"https://(?'host'[-.\w]+)/@(?'name'\w+)/(?'id'\d+)"#`. That code looks like it creates a `url` attribute inside the `extract` definition.

I should mention that `CachedDefinitionCollection.php` is auto-generated from [XML files](https://github.com/s9e/TextFormatter/tree/master/src/Plugins/MediaEmbed/Configurator/sites). The XML files are canonical and easier to read.

Reply:
So... Like the way I just updated it is how it should be?

Reply:
That looks like it. If there's anything wrong with it, I can't see it.